src/tracing/database.rs

```diff
@@ -11,9 +11,9 @@ use tokio::runtime::Handle;
 
 #[derive(Debug)]
 #[allow(clippy::redundant_pub_crate)]
-pub(crate) struct EthDatabaseSnapshot<P: EthereumProvider + Send + Sync> {
-    cache: CacheDB<P>,
-    block_id: BlockId,
+pub struct EthDatabaseSnapshot<P: EthereumProvider + Send + Sync> {
```

Comment:
Here I didn't want to move all that in this PR as it is already quite big but I think that in a follow up PR it will be necessary to remove this snapshot database file from the tracing to put an equivalent file in the provider given that we is used for the state overrides of this snapshot.